addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,194 @@
+    private static HttpSender sender =
+            new HttpSender(Model.getSingleton().getOptionsParam().getConnectionParam(), true, 17);
+    // private static HttpRequestConfig config =
+    // HttpRequestConfig.builder().setSoTimeout(10).build();
+
+    /**
+     * Returns a List of all parameters to be used from a given wordlist file.
+     *
+     * @param path the path to the wordlist.
+     * @return a List<String> of all parameters to be used from a given wordlist file.
+     */
+    public static List<String> read(String path) {
+        logger.info("Reading keys from file: {}", path);
+        List<String> params = new ArrayList<>();
+        try (BufferedReader br = Files.newBufferedReader(Paths.get(path), StandardCharsets.UTF_8)) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                params.add(StringUtils.strip(line));
+            }
+            logger.info("Succesfully parsed keys!");
+        } catch (Exception e) {
+            logger.error(e);
+        }
+        return params;
+    }
+
+    /**
+     * Returns "crashproof" HTTP Responses from a given URL by using GET method.
+     *
+     * @param url the url to be tested.
+     * @param headers the associated headers.
+     * @return a HttpMessage of the HTTP Response from the given URL.
+     */
+    public static HttpMessage stableRequest(String url, HttpRequestHeader headers) {
+        try {
+            URI parsed = new URI(url, true);
+            String scheme = parsed.getScheme();
+            String host = parsed.getHost();
+            String path = parsed.getPath();
+            String port = parsed.getPort() == -1 ? "" : Integer.toString(parsed.getPort());
+            String[] schemes =
+                    (scheme.equalsIgnoreCase("https"))
+                            ? new String[] {"https", "http"}
+                            : new String[] {"http", "https"};
+            int errorCount = 0;
+            for (String s : schemes) {
+                try {
+                    HttpMessage msg = new HttpMessage();
+                    headers.setURI(new URI(s + "://" + host + ":" + port + path, true));
+                    headers.setMethod("GET");
+                    msg.setRequestHeader(headers);
+                    sender.sendAndReceive(msg);
+                    return msg;
+                } catch (Exception e) {
+                    // Check if it is a connection error
+                    if (e.getMessage().contains("ConnectionError")
+                            || e.getMessage().contains("IOException")) {
+                        continue;
+                    }
+                    logger.info(e);
+                    errorCount = errorCount + 1;
+                }
+            }
+            logger.info("Total number of Errors: " + errorCount);
+        } catch (URIException e) {
+            logger.error("Error parsing URL: {} Exception raised {}", url, e);
```

Comment:
I believe this can be simply:
`             logger.error("Error parsing URL: {}", url, e);`